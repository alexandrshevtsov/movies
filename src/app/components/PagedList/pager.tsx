import * as React from 'react';
import { Link } from 'react-router-dom';

interface IPagerProps {
  batchIndex: number;
  batchCount: number;
  className?: string;
  itemClassName?: string;
  pageLink?: string;
  pageClick?: (batchIndex: number) => void;
}

export const Pager = (props: IPagerProps) => {
  const batchCount = props.batchCount;
  const pageLink = props.pageLink;
  if (batchCount <= 1 || !pageLink || pageLink.length === 0) return null;
  let pages: any[] = [];
  for (let batchIndex = 0; batchIndex < batchCount; batchIndex++) {
    const itemKey = `pager-item-${batchIndex}`;
    const link = batchIndex > 0 ? pageLink + '/' + (batchIndex + 1).toString() : pageLink;
    const pageTitle = batchIndex + 1;
    let pageClick = props.pageClick;
    pages.push(
      <li className={props.itemClassName} key={itemKey}>
        {batchIndex === props.batchCount - 1 ? 'of ' : ''}
        {batchIndex === props.batchIndex ? (
          pageTitle
        ) : (
          <Link
            to={link}
            onClick={() => {
              if (pageClick) pageClick(batchIndex);
            }}
          >
            {pageTitle}
          </Link>
        )}
      </li>
    );
  }
  return <ul className={props.className}>{pages}</ul>;
};
