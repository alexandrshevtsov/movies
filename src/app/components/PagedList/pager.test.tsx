import * as React from 'react';
import { mount } from 'enzyme';
import { Pager } from './pager';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Pager', () => {
  it('rendering and navigation', () => {
    const pageClick = jest.fn();
    const component = mount(
      <Router>
        <Pager
          batchCount={3}
          batchIndex={1}
          className="myPager"
          itemClassName="myPagerItem"
          pageLink="/test"
          pageClick={pageClick}
        />
      </Router>
    );
    const pager = component.find('ul');
    expect(pager.length).toBe(1);
    expect(pager.hasClass('myPager')).toBe(true);
    const pages = component.find('li');
    expect(pages.length).toBe(3);
    pages
      .first()
      .find('a')
      .simulate('click');
    expect(pages.first().hasClass('myPagerItem')).toBe(true);
    expect(pageClick).toBeCalledTimes(1);
    expect(pageClick).toBeCalledWith(0);
    expect(component).toMatchSnapshot();
  });
});
