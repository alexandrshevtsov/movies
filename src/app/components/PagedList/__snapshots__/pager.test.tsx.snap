// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager rendering and navigation 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      batchCount={3}
      batchIndex={1}
      className="myPager"
      itemClassName="myPagerItem"
      pageClick={
        [MockFunction] {
          "calls": Array [
            Array [
              0,
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      pageLink="/test"
    >
      <ul
        className="myPager"
      >
        <li
          className="myPagerItem"
          key="pager-item-0"
        >
          <Link
            onClick={[Function]}
            to="/test"
          >
            <a
              href="/test"
              onClick={[Function]}
            >
              1
            </a>
          </Link>
        </li>
        <li
          className="myPagerItem"
          key="pager-item-1"
        >
          2
        </li>
        <li
          className="myPagerItem"
          key="pager-item-2"
        >
          of 
          <Link
            onClick={[Function]}
            to="/test/3"
          >
            <a
              href="/test/3"
              onClick={[Function]}
            >
              3
            </a>
          </Link>
        </li>
      </ul>
    </Component>
  </Router>
</BrowserRouter>
`;
